returnn/torch/engine.py

```diff
@@ -155,6 +186,8 @@ def train(self):
             self.epoch += 1
             self._epoch_mp_shared.value = self.epoch
 
+        if self._use_torch_distributed:
+            destroy_process_group()
```

Comment:
Why are you calling this here? As far as I understood, this function should be called at the end of the process? E.g. in our `finalize` function.